webapp/src/js/services/format-data-record.js

```diff
@@ -55,35 +55,37 @@ angular
 
     const fieldsToHtml = function(
       settings,
```

Comment:
As far as I can tell, this function only runs for SMS reports. 
Even if an XML report had a `case_id` (generated with the transition, so not a `field)`, it wouldn't be displayed. Should this support be added?

Reply:
I don't think this comment is on the right line, but I see what you mean. This happens with `patient_id` too. I've made a change to include these fields as well, but it's a bit messy because the JSON and XML code is so different. Let me know what you think...

Reply:
Awesome, thanks! I think it's good to support generated fields in XML forms as well! 
I also think you cleaned up the template so it's less messy.